src/auth/auth.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable } from '@nestjs/common';
+import { v4 as uuid } from 'uuid';
+import { sign, verify } from 'jsonwebtoken'
+import { compare } from 'bcrypt'
+
+import { User } from '../users/entities/user.entity';
+import { UsersService } from '../users/users.service';
+import { LoginResponseDto } from './dtos/login-response.dto';
+import { RefreshToken } from './entities/refresh-token.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+@Injectable()
+export class AuthService {
+
+    private retrieveRefreshToken(
+        refreshStr: string,
+    ): Promise<RefreshToken | undefined> {
+        try {
+            const decoded = verify(refreshStr, process.env.JWT_REFRESH_SECRET);
+            if (typeof decoded === 'string') {
+                return undefined;
+            }
+            return this.repository.findOne({
+                where: {
+                    id: decoded.id
+                }
+            });
+        } catch (e) {
+            return undefined;
+        }
+    }
+
+    private async newRefreshAndAccessToken(
+        user: User,
+        userAgent: string,
+        ipAddress: string
+    ): Promise<LoginResponseDto> {
+        const refreshToken = new RefreshToken({
+            userAgent,
+            ipAddress,
+            userId: user.id
+        });
+
+        this.repository.save(refreshToken);
+
+        return {
+            refreshToken: refreshToken.sign(),
+            accessToken: sign({ userId: user.id }, process.env.JWT_SECRET, { expiresIn: '1h' })
```

Comment:
Can we add username and email to JWT token as well, since we have that information and can be useful?